cpp/src/packed/reader.cpp

```diff
@@ -49,15 +49,11 @@ PackedRecordBatchReader::PackedRecordBatchReader(arrow::fs::FileSystem& fs,
     file_readers_.emplace_back(std::move(result.value()));
   }
 
-  auto parse_size = [&](const std::string& input) -> std::vector<int64_t> {
-    std::vector<int64_t> sizes;
-    std::string token;
-    std::stringstream ss(input);
-
-    while (std::getline(ss, token, ',')) {
-      sizes.push_back(std::stoll(token));
-    }
-    return sizes;
+  auto parse_size = [&](const std::string& input) -> std::vector<size_t> {
+    std::vector<uint8_t> byteArray = std::vector<uint8_t>(input.begin(), input.end());
+    std::vector<size_t> vec(byteArray.size() / sizeof(size_t));
+    std::memcpy(vec.data(), byteArray.data(), byteArray.size());
+    return vec;
   };
   for (int i = 0; i < file_readers_.size(); ++i) {
     row_group_sizes_.push_back(
```

Comment:
`->value(0)`: don't do that, the `key_value_metadata` field may have multiple entries.